rpc/codec/server.go

```diff
@@ -79,6 +91,58 @@ func (c *serverCodec) ReadRequestHeader(r *rpc.Request) error {
 	return nil
 }
 
+// authenticateRequest takes a request proto and attempts to authenticate it.
+// For requests with a RequestHeader, we compare the header.User against
+// the client certificate Subject.CommonName.
+// For other requests, we either allow all (test requests), or
+// require the "node user".
+func (c *serverCodec) authenticateRequest(request proto.Message) error {
+	var err error
+	var requestedUser string
+
+	switch request := request.(type) {
+	case cockroach_proto.Request:
```

Comment:
Good point. I'll settle for that in the short term. Long-term, we need to see if there are more things we may need from the request header.